### ISSUE TITLE
Add InitializeOpenFgaAutoconfiguration

### ISSUE THREAD

**Issue by rwinch** [jimmyjames/fga-spring-examples#5]
Similar to Spring Boot's support for [initializing a database](https://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#howto.data-initialization.using-basic-sql-scripts) it would be nice to add support for initializing OpenFGA using convention based resources for both the authorization model(s) and initial relations.

This is very similar to `LoadFgaData` except:

- It is automatically activated based upon the presence of well known files on the classpath. Users would only need to add files in the well known location(s) to activate the configuration
- We can add support for customizing the location
- We could take this further and create a (non-spring) library similar to [liquibase](https://www.liquibase.com/) could be created for managing the configuration in production (possibly separate ticket)
- We can consider adding a way to specify data at the test method level  (possibly separate ticket)

**Comment by Crain-32** [comment]
I got a version working locally (for the most part), just waiting on the Testcontainer PR to really test it, then I can make another PR.

It also adds in a new property `fgaStoreName`, since we should assume if we're populating the auth model, we need to create a Store. I've included the code for reference since I can't make a PR yet.
```java
    private void doInitialization() throws Exception {
        if (openFgaProperties.getStoreId() != null) {
            validateStore(openFgaProperties.getFgaStoreId());
        } else {
            openFgaProperties.setFgaStoreId(makeStore());
        }

        if (openFgaProperties.getInitScript() != null) {
            var script = resourceLoader.getResource(openFgaProperties.getInitScript());
            var mapper = new ObjectMapper(); // Wrong thing to do for a proper init, but this works for now.
            var authWriteResponse = openFgaClient.writeAuthorizationModel(mapper.readValue(script.getContentAsByteArray(), WriteAuthorizationModelRequest.class)).get();
            openFgaProperties.setFgaAuthorizationModelId(authWriteResponse.getAuthorizationModelId());
        }
 ```

**Comment by Crain-32** [comment]
@rwinch I see you doing some PR stuff. Are you adding this to your PR? I was just about to push my implementation before checking up on it.

I also broke the gradlew in gh-6, that's on me

**Comment by rwinch** [comment]
@Crain-32

> Looking at the PR right now, I'll base my changes off that, and just keep an eye on any changes you make.

Sorry about that. I hope that it didn't make your changes too difficult. I've finished with a single PR with multiple commits to make it easier to merge.

> I also broke the gradlew in https://github.com/jimmyjames/fga-spring-examples/issues/6, that's on me

No worries. It should be fixed now.

**Comment by Crain-32** [comment]
None of your changes impacted me, since I'm just targeting the `LoadFgaData` for my stuff.

Will make rebasing on your changes a lot easier, should that happen.
